### PR TITLE
Box documentation

### LINKED ISSUE #1727: Box Documentation

### PR DESCRIPTION
**Pull request by parth-elastic**
## Closes #1727 

This is a documentation PR for Box connector.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Considered corresponding documentation changes

### CODE REVIEW

File: docs/reference/box.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+For additional operations, see [Usage](https://www.elastic.co/guide/en/enterprise-search/master/connectors-usage.html).
+
+## Box API Authorization
+
+### Box Free Account
+
+#### Create Box User Authentication (OAuth 2.0) Custom App
+
+You'll need to create an User Authentication (OAuth 2.0) app in the Box developer console by following these steps:
```
Comment (leemthompo) [inline docs/reference/box.md:23]:
```suggestion
You'll need to create an OAuth app in the Box developer console by following these steps:
```

File: docs/reference/box.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+1. Register a new app in the [Box dev console](https://app.box.com/developers/console) with custom App and select User authentication (OAuth 2.0).
+
+2. Add elasticsearch index URL in Redirect URIs
```
Comment (leemthompo) [inline docs/reference/box.md:27]:
```suggestion
2. Add your Elasticsearch endpoint URL to **Redirect URIs**.
```

File: docs/reference/box.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+3. Check "Write all files and folders stored in Box" in Application Scopes.
+
+4. Once the app is created, users can have Client ID and Client secret on the configuration tab.
```
Comment (leemthompo) [inline docs/reference/box.md:31]:
```suggestion
4. Once the app is created, **Client ID** and **Client secret** values are available in the configuration tab. Keep these handy.
```

File: docs/reference/box.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+#### Generate a refresh Token
+
+To generate a refresh token, follow these steps:
+
+1. Go to the following URL, replacing `<CLIENT_ID>` with the **Client ID** value saved earlier.
+    ```shell
+    "https://account.box.com/api/oauth2/authorize?response_type=code&client_id=<CLIENT_ID>"
```
Comment (leemthompo) [inline docs/reference/box.md:39]:
```suggestion
    https://account.box.com/api/oauth2/authorize?response_type=code&client_id=<CLIENT_ID>
```

Don't think need quotes here?

File: docs/reference/box.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    ```
+    
+The HTTP response should contain an **authorization code** that you'll use to generate a refresh token.
+
+**Note:** Authorization code can only be used once to create a refresh token and itâ€™s valid for 30 seconds only.
```
Comment (leemthompo) [inline docs/reference/box.md:44]:
```suggestion
**Note:** Authorization codes to generate refresh tokens can only be used once and are only valid for 30 seconds.
```

File: docs/reference/box.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+2. In your terminal, run the following `cURL` command, replacing `<AUTHORIZATION_CODE>`, `<CLIENT_ID> and <CLIENT_SECRET>` with the values you saved earlier:
```
Comment (leemthompo) [inline docs/reference/box.md:46]:
```suggestion
2. In your terminal, run the following `curl` command, replacing `<AUTHORIZATION_CODE>`, `<CLIENT_ID> and <CLIENT_SECRET>` with the values you saved earlier:
```

File: docs/reference/box.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+    ```shell
+    curl -i -X POST "https://api.box.com/oauth2/token" \
+        -H "Content-Type: application/x-www-form-urlencoded" \
+        -d "client_id=<CLIENT_ID>" \
+        -d "client_secret=<CLIENT_SECRET>" \
+        -d "code=<AUTHORIZATION_CODE>" \
+        -d "grant_type=authorization_code"
+    ```
+Store the refresh token from the response and use it in the connector configuration.
```
Comment (leemthompo) [inline docs/reference/box.md:55]:
```suggestion
Save the refresh token from the response. You'll need this for the connector configuration.
```

File: docs/reference/box.md [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+1. Register a new app in the [Box dev console](https://app.box.com/developers/console) with custom App and select Server Authentication (Client Credentials Grant).
+
+2. Check following permissions:
+    - "Write all files and folders stored in Box" in Application Scopes
+    - "Make API calls using the as-user header" in Advanced Features
+
+3. Select `App + Enterprise Access` in App Access Level.
+
+4. Then authorized your application by admin console.
```
Comment (leemthompo) [inline docs/reference/box.md:69]:
```suggestion
4. Authorize your application from the admin console.
```

File: docs/reference/box.md [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+
+Store the Client Credentials and Enterprise ID and use it in the connector configuration.
```
Comment (leemthompo) [inline docs/reference/box.md:71]:
```suggestion
Save the **Client Credentials** and **Enterprise ID**. You'll need these to configure the connector.
```

File: docs/reference/box.md [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+
+#### `Client Secret`  (required)
+
+The Client Secret to authenticate with Box instance.
+
+#### `Refresh Token` (required if Box Account is Box Free)
+
+The Refresh Token to generate Access Token. 
+
+**NOTE:** If the process terminates, then user needs to generate a new refresh token and configure
```
Comment (leemthompo) [inline docs/reference/box.md:93]:
```suggestion
**NOTE:** If the process terminates, you'll need to generate a new refresh token.
```

**Review by leemthompo — COMMENTED** [review]
Some minor cleanup writing wise, over to Dmitrii for the technical approval

File: docs/reference/box.md [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+
+### Box Free Account
+
+#### Create Box User Authentication (OAuth 2.0) Custom App
+
+You'll need to create an OAuth app in the Box developer console by following these steps:
+
+1. Register a new app in the [Box dev console](https://app.box.com/developers/console) with custom App and select User authentication (OAuth 2.0).
+
+2. Add your Elasticsearch endpoint URL to **Redirect URIs**.
```
Comment (vidok) [inline docs/reference/box.md:27]: Why Elasticsearch endpoint?

Comment (abhishekjoshi-crest) [reply docs/reference/box.md:27]: because it will redirect to user where he/she needs to configure the connector.

Comment (parth-elastic) [reply docs/reference/box.md:27]: Because it will redirect the user to where he or she needs to configure the connector.

Comment (vidok) [reply docs/reference/box.md:27]: I'm afraid the Elasticsearch endpoint can't be an HTML page. Elasticsearch exposes only API afaik. 

Also, how can the customer grab an `authorization code` if they are redirected to another page? Usually, a page should handle the `authorization code` and use it.

Comment (parth-elastic) [reply docs/reference/box.md]: Okay then I have removed this step and now user will redirect to default location

Comment (vidok) [reply docs/reference/box.md]: Can you please describe how customers can find `authorization code`? 

As far as I understand, the authorization code is a part of the redirect payload so the customers should be able to see it. We need to list those steps. 

I was able to grab the code only when I created a local web server and used localhost as a redirect page. Do you have other options?

Comment (parth-elastic) [reply docs/reference/box.md]: I have updated the doc to find `authorization code`

File: docs/reference/box.md [hunk 12]
```diff
@@ -36,8 +38,10 @@ To generate a refresh token, follow these steps:
     ```shell
     https://account.box.com/api/oauth2/authorize?response_type=code&client_id=<CLIENT_ID>
     ```
-    
-The HTTP response should contain an **authorization code** that you'll use to generate a refresh token.
+
+2. Grant access to your application.
+
+3. Now you will redirect to the web page that you configured in **Redirect URIs**, and the HTTP response should contain an **authorization code** that you'll use to generate a refresh token.
```
Comment (vidok) [inline docs/reference/box.md:44]:
```suggestion
3. You will now be redirected to the web page that you configured in **Redirect URIs**, and the HTTP response should contain an **authorization code** that you'll use to generate a refresh token.
```

**Review by vidok — APPROVED** [review]

**Pull request merged**